[Migrate] Kotlin Android Extensions to ViewBinding

ParaskP7 opened the pull request:
Depends On:
 - [x] [ADC#32](https://github.com/Automattic/android-dependency-catalog/pull/32)
 - [x] Update ADC to `1.17.0`

### Description

This PR is about migrating from the deprecated `kotlin-android-extensions` plugin, replacing Kotlin synthetics with ViewBinding and using `kotlin-parcelize` instead.

FYI: The `kotlin-android-extensions` plugin has been deprecated for years now and it is now preventing us from updating to a newer Kotlin version (ie. [Kotlin 1.9.25](https://github.com/JetBrains/kotlin/releases/tag/v1.9.25)).

### To Test

- Smoke test the `FluxC Example` app, try out all available screens and functionality. Verify everything is working as expected.
- Also, if you want to be thorough about reviewing the `kotlinx.parcelize.Parcelize` import change, you could quickly smoke test, either the [JP/WPAndroid](https://github.com/wordpress-mobile/WordPress-Android) and/or [WCAndroid](https://github.com/woocommerce/woocommerce-android), with this version of `FluxC`, or via composite builds, and see if it works as expected.

ParaskP7 committed:
Build: Replace kotlin android extensions with parcelize for example

As part of this, the deprecated parcelize import got also replaced.

ParaskP7 committed:
Refactor: Replace account fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace domains fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace editor theme fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace experiments fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace jetpack ai fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace main fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace notifications fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace plans fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace plugins fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace posts fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace react native fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace theme fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo coupons fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo update coupon fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo customers fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo customer creation fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo customers search builder synthetics with binding

ParaskP7 committed:
Refactor: Replace woo customers search fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo gateways fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo help support fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo leaderboards fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo onboarding fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace address edit dialog fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo orders fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace addons test fragment synthetics with binding

ParaskP7 committed:
Fix: Add missing on view created super call on woo batch generate v

ParaskP7 committed:
Refactor: Replace woo batch generate v fragment synthetics with binding

FYI: This refactor also fixes a few bugs where a view id from
'fragment_woo_batch_update_variations' was used instead of using the
correct with, with the same view id, from the actual targeted
'fragment_woo_batch_generate_variations' fragment.

ParaskP7 committed:
Fix: Add missing on view created super call on woo batch update v

ParaskP7 committed:
Refactor: Replace woo batch update v fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo product attribute fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo product categories synthetics with binding

ParaskP7 committed:
Refactor: Replace woo product downloads fragment synthetics with binding

FYI: This refactor also fixes a bug where a view id from
'fragment_woo_product_categories' was used instead of using the
correct with, with the same view id, from the actual targeted
'fragment_woo_product_downloads' fragment.

ParaskP7 committed:
Fix: Add missing on view created super call on woo product filters

ParaskP7 committed:
Refactor: Replace woo product filters fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo product tags fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo products fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo update product fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo update variation fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo refunds fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo shipping label fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo verify address fragment synthetics with binding

FYI: This refactor also fixes a bug where a view id from
'fragment_woo_shippinglabels' was used instead of using the
correct with, with the same view id, from the actual targeted
'fragment_woo_verify_address' fragment.

ParaskP7 committed:
Refactor: Replace woo revenue stats fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo store creation fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo tax fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace woo admin fragment synthetics with binding

ParaskP7 committed:
Refactor: Replace main example activity synthetics with binding

ParaskP7 committed:
Refactor: Replace wc order list activity synthetics with binding

ParaskP7 committed:
Refactor: Replace woo commerce fragment synthetics with binding

ParaskP7 committed:
Fix: Remove wrongly added synthetics and its button container usage

ParaskP7 committed:
Cleanup: Remove unused custom stats dialog and its synthetics usage

ParaskP7 committed:
Refactor: Replace notification type subtype synthetics with binding

ParaskP7 committed:
Refactor: Replace signin dialog synthetics with binding

ParaskP7 committed:
Refactor: Replace wc add order shipment tracking synthetics with binding

ParaskP7 committed:
Refactor: Replace floating label edit text synthetics with binding

ParaskP7 committed:
Refactor: Replace woo customers search adapter synthetics with binding

ParaskP7 committed:
Refactor: Replace woo product categories adapter synthetics with binding

ParaskP7 committed:
Refactor: Replace woo product downloads adapter synthetics with binding

ParaskP7 committed:
Refactor: Replace woo product tag adapter synthetics with binding

ParaskP7 committed:
Analysis: Remove wildcard import exclusion from detekt config

ParaskP7 committed:
Build: Remove kotlin android extensions config from example module

ParaskP7 committed:
Build: Remove kotlin android extensions config from fluxc module

ParaskP7 committed:
Build: Replace kotlin android extensions with parcelize for fluxc

As part of this, the deprecated parcelize import got also replaced.

ParaskP7 committed:
Build: Migrate to kotlin parcelize (via catalog's pr-sha combination)

FYI: This change was done for testing purposes and until the below
'Android Dependency Catalog' #32 PR gets merged to 'trunk'. When that's
done, the 'catalogVersion' will be updated to '1.17.0' instead.

------------------------------------------------------------------------

Android Dependency Catalog PR: [Migrate] Kotlin Android Extensions to
Kotlin Parcelize #32
- https://github.com/Automattic/android-dependency-catalog/pull/32

ParaskP7 committed:
Analysis: Suppress complex method warning on theme fragment

ParaskP7 committed:
Analysis: Remove redundant sam constructor on woo update fragments

ParaskP7 committed:
Analysis: Resolve max line length warnings for example module

wzieba reviewed (approved):
Looks good, great work @ParaskP7 🙌 . Thanks for improving some log messages along the way.

btw: during the review, I started to wonder how often the `example` app in FluxC is used nowadays. Maybe it's something we could discuss internally what's the future for this module.

Not merging as I understand we need to update versions catalog 👍

ParaskP7 commented:
> Looks good, great work @ParaskP7 🙌 . Thanks for improving some log messages along the way.

Awesome, thanks for the quick review @wzieba ! 🙇 ❤️ 

> btw: during the review, I started to wonder how often the example app in FluxC is used nowadays.

Hmmm, to my understanding, to some extend, but to be honest, I am actually not sure. I bet sometimes it might help to use this `example` app, to speed-up development and testing, while other times, using included builds might be preferred... 🤷 

We could ask JP/WPAndroid and WCAndroid devs this question directly and get a quick feedback on it. 🤔 

> Maybe it's something we could discuss internally what's the future for this module.

👍 💯 

> Not merging as I understand we need to update versions catalog 👍

Yes, thanks you, I am in the process of creating the new [1.17.0](https://github.com/Automattic/android-dependency-catalog/releases/tag/1.17.0) release, will then push an update commit and merge it myself. 👍

ParaskP7 committed:
Deps: Update catalog to 1.17.0 (kotlin parcelize migration)

Catalog: https://github.com/Automattic/android-dependency-catalog/
releases/tag/1.17.0
